flutter/lib/src/integrations/on_error_integration.dart

```diff
@@ -0,0 +1,123 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+import 'package:sentry/sentry.dart';
+import '../sentry_flutter_options.dart';
+
+typedef ErrorCallback = bool Function(Object exception, StackTrace stackTrace);
+
+/// Integration which captures `PlatformDispatcher.onError`
+/// See:
+/// - https://master-api.flutter.dev/flutter/dart-ui/PlatformDispatcher/onError.html
+///
+/// Remarks:
+/// - Only usable on Flutter >= 3.1.0
+///
+/// This can be used instead of the [RunZonedGuardedIntegration]. Removing the
+/// [RunZonedGuardedIntegration] results in a minimal improved startup time,
+/// since creating [Zone]s is not cheap.
+class OnErrorIntegration implements Integration<SentryFlutterOptions> {
+  OnErrorIntegration({this.dispatchWrapper});
+
+  ErrorCallback? _defaultOnError;
+  ErrorCallback? _integrationOnError;
+  PlatformDispatcherWrapper? dispatchWrapper;
+  SentryFlutterOptions? _options;
+
+  @override
+  void call(Hub hub, SentryFlutterOptions options) {
+    _options = options;
+    final wrapper = dispatchWrapper ??
+        // WidgetsBinding works with WidgetsFlutterBinding and other custom bindings
+        PlatformDispatcherWrapper(WidgetsBinding.instance.platformDispatcher);
+
+    if (!wrapper.isOnErrorSupported(options)) {
+      return;
+    }
+    _defaultOnError = wrapper.onError;
+
+    _integrationOnError = (Object exception, StackTrace stackTrace) {
+      final handled = _defaultOnError?.call(exception, stackTrace) ?? false;
```

Comment:
The default value could also be `true`? What do you guys think?

Reply:
Hard to say, I'd need to test it out a bit, but I guess by default should be `true` for now due to the app never crashing anyways.